templates/todo/projects/python-mongo/.repo/terraform/infra/main.tf

```diff
@@ -0,0 +1,306 @@
+# Deploy app service api
+# ------------------------------------------------------------------------------------------------------
+resource "azurecaf_name" "appi_name" {
+  name          = "api-${random_string.resource_token.result}"
+  resource_type = "azurerm_app_service"
+  random_length = 0
+  clean_input   = true
+}
+
+
+resource "azurerm_linux_web_app" "api" {
+  name                = azurecaf_name.appi_name.result
+  location            = azurerm_resource_group.rg.location
+  resource_group_name = azurerm_resource_group.rg.name
+  service_plan_id     = azurerm_service_plan.plan.id
+  https_only          = true
+  tags                = merge(local.tags, { "azd-service-name" : "api" })
+
+  site_config {
+    #linux_fx_version = "PYTHON|3.8"
+    always_on        = true
+    ftps_state       = "FtpsOnly"
+    app_command_line = local.api_command_line
+  }
+
+  identity {
+    type = "SystemAssigned"
+  }
+
+  app_settings = {
+    "AZURE_COSMOS_CONNECTION_STRING_KEY"    = "AZURE-COSMOS-CONNECTION-STRING"
+    "AZURE_COSMOS_DATABASE_NAME"            = azurerm_cosmosdb_mongo_database.mongodb.name
+    "SCM_DO_BUILD_DURING_DEPLOYMENT"        = "true"
+    "AZURE_KEY_VAULT_ENDPOINT"              = azurerm_key_vault.kv.vault_uri
+    "APPLICATIONINSIGHTS_CONNECTION_STRING" = module.applicationinsights.APPLICATIONINSIGHTS_CONNECTION_STRING
+  }
+  logs {
+    application_logs {
+      file_system_level = "Verbose"
+    }
+    detailed_error_messages = true
+    failed_request_tracing  = true
+    http_logs {
+      file_system {
+        retention_in_days = 1
+        retention_in_mb   = 35
+      }
+    }
+  }
+}
+
+# ------------------------------------------------------------------------------------------------------
+# Deploy log analytics workspace
+# ------------------------------------------------------------------------------------------------------
+resource "azurecaf_name" "workspace_name" {
+  name          = random_string.resource_token.result
+  resource_type = "azurerm_log_analytics_workspace"
+  random_length = 0
+  clean_input   = true
+}
+
+resource "azurerm_log_analytics_workspace" "workspace" {
+  name                = azurecaf_name.workspace_name.result
+  location            = azurerm_resource_group.rg.location
+  resource_group_name = azurerm_resource_group.rg.name
+  sku                 = "PerGB2018"
+  retention_in_days   = 30
+  tags                = local.tags
+}
+
+# ------------------------------------------------------------------------------------------------------
+# Deploy cosmos db account
+# ------------------------------------------------------------------------------------------------------
+resource "azurecaf_name" "db_acc_name" {
+  name          = random_string.resource_token.result
+  resource_type = "azurerm_cosmosdb_account"
+  random_length = 0
+  clean_input   = true
+}
+
+resource "azurerm_cosmosdb_account" "db" {

```

Comment:
Thoughts on whether to leave all the resources in one `main.tf` file vs break them out into separate files per resource? Any time I have done TF development in the past I found it easier to maintain if the resources were split up into multiple files.

Reply:
@wbreza  I broke down main.tf into multiple file.